common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -334,6 +350,167 @@ class NativeAuthMsalController : BaseNativeAuthController() {
+                }
+                is SignUpStartApiResult.InvalidPassword -> {
+                    SignUpCommandResult.InvalidPassword(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription
+                    )
+                }
+                is SignUpStartApiResult.InvalidAttributes -> {
+                    SignUpCommandResult.InvalidAttributes(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription,
+                        invalidAttributes = signUpStartApiResult.invalidAttributes
+                    )
+                }
+                is SignUpStartApiResult.UsernameAlreadyExists -> {
+                    SignUpCommandResult.UsernameAlreadyExists(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription
+                    )
+                }
+                is SignUpStartApiResult.InvalidEmail -> {
+                    SignUpCommandResult.InvalidEmail(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription
+                    )
+                }
+                is SignUpStartApiResult.AuthNotSupported -> {
+                    SignUpCommandResult.AuthNotSupported(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription
+                    )
+                }
+                is SignUpStartApiResult.Redirect -> {
+                    INativeAuthCommandResult.Redirect()
+                }
+                is SignUpStartApiResult.UnsupportedChallengeType, is SignUpStartApiResult.UnknownError -> {
+                    signUpStartApiResult as ApiErrorResult
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result: $signUpStartApiResult"
+                    )
+                    INativeAuthCommandResult.UnknownError(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription
+                    )
+                }
+            }
+        } catch (e: Exception) {
+            Logger.error(TAG, "Exception thrown in signUpStart", e)
+            throw e
+        }
+    }
+
+    /**
+     * Makes a call to the signup/continue endpoint using the provided code, mapping responses returned from the server into a command result.
+     */
+    fun signUpSubmitCode(parameters: SignUpSubmitCodeCommandParameters): SignUpSubmitCodeCommandResult {
+        LogSession.logMethodCall(TAG, "${TAG}.signUpSubmitCode")
+
+        try {
+            val oAuth2Strategy = createOAuth2Strategy(parameters)
+
+            val signUpContinueApiResult = performSignUpSubmitCode(
```

Comment:
`signUpContinueApiResult` rename to `signUpSubmitCodeResult`

Reply:
Done